import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+        "meta_mRNA_seq_fpkm_capture.txt",
+        "meta_mRNA_seq_fpkm_capture_Zscores.txt",
+        "meta_mRNA_seq_fpkm_capture_all_sample_Zscores.txt",
+        "meta_mRNA_seq_fpkm_polya.txt",
+        "meta_mRNA_seq_fpkm_polya_Zscores.txt",
+        "meta_mRNA_seq_fpkm_polya_all_sample_Zscores.txt",
+        "meta_methylation_binary.txt",
+        "meta_methylation_genebodies_hmEPIC.txt",
+        "meta_methylation_genebodies_wgbs.txt",
+        "meta_methylation_hm27.txt",
+        "meta_methylation_hm450.txt",
+        "meta_methylation_promoters_hmEPIC.txt",
+        "meta_methylation_promoters_wgbs.txt",
+        "meta_miRNA_median_Zscores.txt",
+        "meta_microbiome.txt",
+        "meta_mrnaseq_fcount.txt",
+        "meta_mrnaseq_rsem.txt",
+        "meta_mutational_signature_contribution_v2.txt",
+        "meta_mutational_signature_contribution_v3.txt",
+        "meta_mutational_signature_pvalue_v2.txt",
+        "meta_mutational_signature_pvalue_v3.txt",
+        "meta_mutations_extended.txt",
+        "meta_mutations_uncalled.txt",
+        "meta_mutsig.txt",
+        "meta_phosphoprotein_quantification.txt",
+        "meta_protein_quantification.txt",
+        "meta_protein_quantification_Zscores.txt",
+        "meta_protein_quantification_zscores.txt",
+        "meta_resource_definition.txt",
+        "meta_resource_sample.txt",
+        "meta_resource_study.txt",
+        "meta_rppa.txt",
+        "meta_rppa_Zscores.txt",
+        "meta_timeline.txt"
+        ]
+    filenames_in_use_before_update_set = set()
+    wildcard_patterns_in_use_before_update_set = set()
+    for filename in FILENAMES_IN_USE_BEFORE_UPDATE_LIST:
+        if "*" in filename:
+            wildcard_patterns_in_use_before_update_set.add(filename.replace(CANCER_STUDY_TAG(), cancer_study_identifier))
+        else:
+            filenames_in_use_before_update_set.add(filename.replace(CANCER_STUDY_TAG(), cancer_study_identifier))
+    return filenames_in_use_before_update_set, wildcard_patterns_in_use_before_update_set
+
+def parse_current_datatypes(cancer_study_identifier):
+    """
+    the contents of the datatype sheet (in tab separated format) are parsed and all filenames (both meta and staging) or wildcard patterns are mapped to their associated datatypes. The maps are returned.
+    """
+    datatype_sheet_lines = datatype_sheet_download_tab_separated().splitlines()
+    filename2datatype = {}
+    wildcard_pattern2datatype = {}
+    header_skipped = False
+    for line in datatype_sheet_lines:
+        if not header_skipped:
+            if line.startswith("datatype"):
+                header_skipped = True
+            continue
+        fields = line.split("\t")
+        datatype = fields[0].strip()
+        staging_filename = fields[5].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        meta_filename = fields[9].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        if "*" in staging_filename:
```

Comment:
could we add a couple examples of these into the comment above? e.g (blah_* would become blah)

Reply:
👍

Reply:
I've added the newly introduced pattern data_mutations_uniprot_canonical_transcripts_*.txt as an example